dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry/internal/rest/TelemetryContainerFilter.java

```diff
@@ -113,17 +128,20 @@ public void filter(final ContainerRequestContext request, final ContainerRespons
         // Note: for async resource methods, this may not run on the same thread as the other filter method
         // Scope is ended in TelemetryServletRequestListener to ensure it does run on the original request thread
 
-        if (instrumenter == null) {
-            return;
-        }
+        try {
+            if (instrumenter == null) {
+                return;
+            }
 
-        Context spanContext = (Context) request.getProperty(SPAN_CONTEXT);
-        if (spanContext == null) {
-            return;
-        }
+            Context spanContext = (Context) request.getProperty(SPAN_CONTEXT);
+            if (spanContext == null) {
+                return;
+            }
 
-        try {
+        
             instrumenter.end(spanContext, request, response, null);
+        } catch (Exception e) {
+            Tr.error(tc, Tr.formatMessage(tc, "CEMOT5101.telemetry.error", e));
         } finally {
             request.removeProperty(REST_RESOURCE_CLASS);
             request.removeProperty(REST_RESOURCE_METHOD);
```

Comment:
Semantics are changed here. Previously the `removeProperty(` lines did not run if `spanContext` or `instrumenter` were `null` and now they do.